lib/classroom_api.rb

```diff
@@ -8,190 +8,229 @@ def initialize(access_token)
-            number_of_api_calls_per_minutes = 0
-            sleep(61.seconds)
-            if redo_loop_number > 9
-              @debug = true
-              @log.api_logger.debug "add_facility_id_to_classrooms, error: API return: #{result['errorcode']} - #{result['error']} after #{number_of_api_calls_per_minutes} calls #{redo_loop_number} times "
-              return @debug
+    begin
+      @rooms_in_db = Room.all.pluck(:rmrecnbr)
+      result = get_classrooms_list
+      if result['success']
+        classrooms_list = result['data']
+        number_of_api_calls_per_minutes = 0
+        redo_loop_number = 1
+        classrooms_list.each do |room|
+          # update only rooms for campuses and buildings from the MClassroom database
+          if @buildings_ids.include?(room['BuildingID'].to_i)
+            if number_of_api_calls_per_minutes < 400
+              number_of_api_calls_per_minutes += 1
+            else
+              @log.api_logger.debug "add_facility_id_to_classrooms, the script sleeps after #{number_of_api_calls_per_minutes} calls"
+              number_of_api_calls_per_minutes = 1
+              sleep(61.seconds)
             end
-            redo_loop_number += 1
-            redo
-          elsif result['success']
-            room_info = result['data'][0]
-            rmrecnbr = room_info['RmRecNbr'].to_i
-            room_in_db = Room.find_by(rmrecnbr: rmrecnbr)
-            if room_in_db
-              if room_in_db.update(facility_code_heprod: facility_id, instructional_seating_count: room_info['RmInstSeatCnt'], campus_record_id: CampusRecord.find_by(campus_cd: room_info['CampusCd']).id)
-                @rooms_in_db.delete(rmrecnbr)
-              else
-                @log.api_logger.debug "add_facility_id_to_classrooms, error: Could not update: rmrecnbr - #{rmrecnbr}, facility_id - #{facility_id}"
+            facility_id = room['FacilityID'].to_s
+            # add facility_id and number of seats
+            result = get_classroom_info(ERB::Util.url_encode(facility_id))
+            if result['errorcode'] == "ERR429"
+              @log.api_logger.debug "add_facility_id_to_classrooms, error: API return: #{result['errorcode']} - #{result['error']} after #{number_of_api_calls_per_minutes} calls"
+              number_of_api_calls_per_minutes = 0
+              sleep(61.seconds)
+              if redo_loop_number > 9
                 @debug = true
+                @log.api_logger.debug "add_facility_id_to_classrooms, error: API return: #{result['errorcode']} - #{result['error']} after #{number_of_api_calls_per_minutes} calls #{redo_loop_number} times "
                 return @debug
               end
+              redo_loop_number += 1
+              redo
+            elsif result['success'] 
+              if result['data']['Classroom'].present?
```

Comment:
Because result['success'] == true doesn't mean that result['data']['Classroom'] has data. It might be empty. If it's not empty then it's an array and the classroom info is in the result['data']['Classroom'][0].
Here is an example:
```{"Classroom"=>[{"FacilityID"=>"400NI1180", "RmRecNbr"=>"2040185", "RmTyp"=>"110", "RmTypDescrShort"=>"Classroom", "RmInstSeatCnt"=>"18", "SchedulingDeptID"=>"440000", "SchedulingDeptDescr"=>"School of Nursing", "BuildingID"=>"1000333", "BldDescrShort"=>"NURSING B1", "BldDescr50"=>"SCHOOL OF NURSING BUILDING 1", "CampusCd"=>"300", "CampusDescr"=>"Medical Campus"}]}```

If the result is empty it is 
`{"Classroom"=>[]}`
 but it's still a successful API call. I have to check if result['data']['Classroom'].present?